tests/test_complexity.py

```diff
@@ -0,0 +1,156 @@
+
+    class ConvModel(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.layers = nn.ModuleList()
+            cin = input_shape[1]
+
+            for cout, stride, kernel, padding in zip(
+                layer_channels, layer_strides, layer_kernels, layer_padding
+            ):
+                self.layers.append(
+                    nn.Conv2d(
+                        cin, cout, kernel_size=kernel, stride=stride, padding=padding
+                    )
+                )
+                cin = cout
+
+        def forward(self, inp):
+            flops = torch.tensor(0)
+            out = inp
+            for layer, k_size in zip(self.layers, layer_kernels):
+                cin = out.shape[1]
+                out = layer(out)
+                # +3 for counting the additions to the flops variable itself
+                flops += out.numel() * k_size * k_size * cin + 3
+
+            return out, flops
+
+    model = ConvModel()
+    inp = torch.randn(input_shape)
+    _, correct_flops = model(inp)
+
+    counted_flops, _, unsupported_ops = ncF.count_flops(model, (inp,))
+    assert torch.allclose(torch.tensor(counted_flops).long(), correct_flops)
+    assert len(unsupported_ops) == 0
+
+
+@pytest.mark.parametrize("input_shape", ((10, 3), (1, 1), (5, 9), (10, 9, 8)))
+def test_flop_count_elementwise(input_shape):
+    # Tests that the flop counter counts flops correctly (i.e.
+    # when compared to a reference value) for a model involving
+    # lots of binary and elementwise operations.
+
+    class ElementwiseModel(nn.Module):
+        def forward(self, inp):
+            flops = torch.tensor(0)
+            out = inp
+
+            out = out * 2
+            # +1 here and below for counting the additions to the flop
+            # variable itself.
+            flops += out.numel() + 1
+
+            out = out.pow(2).exp().log()
+            flops += out.numel() * 3 + 1
+
+            out = out.view(-1, 1) @ out.view(1, -1)
+            flops += out.numel() + 1
+
+            out = out + out
+            flops += out.numel() + 1
+
+            return out, flops
+
+    model = ElementwiseModel()
+    inp = torch.randn(input_shape)
+    _, correct_flops = model(inp)
+
+    counted_flops, _, unsupported_ops = ncF.count_flops(
+        model, (inp,), use_single_flop_estimates=True
+    )
+    assert torch.allclose(torch.tensor(counted_flops).long(), correct_flops)
+    assert len(unsupported_ops) == 0
+
+
+@pytest.mark.parametrize(
+    "batch_size, network_channels, compression_channels, img_size", [(1, 8, 16, 128)]
+)
+def test_flop_count_hyperprior(
+    batch_size, network_channels, compression_channels, img_size
+):
+    # Testing that the flop counter doesn't crash and records all operators
+    # on a more complicated model, i.e. the scale hyperprior model.
+    model = ScaleHyperprior(
+        network_channels=network_channels, compression_channels=compression_channels
+    )
```

Comment:
Do we have any estimates for how many FLOPs this should be? We should verify that our results match other papers